bindings/generator/api/addr.py

```diff
@@ -0,0 +1,92 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+from typing import Optional
+
+from .common import (
+    EntryID,
+    InvitationToken,
+    OrganizationID,
+    Ref,
+    Result,
+    StrBasedType,
+    U32BasedType,
+    Variant,
+    VariantItemUnit,
+)
+
+
+class BackendAddr(StrBasedType):
+    custom_from_rs_string = "|s: String| -> Result<_, String> { libparsec::BackendAddr::from_any(&s).map_err(|e| e.to_string()) }"
+    custom_to_rs_string = "|addr: libparsec::BackendAddr| -> Result<String, &'static str> { Ok(addr.to_url().into()) }"
+
+
+class BackendOrganizationAddr(StrBasedType):
+    custom_from_rs_string = "|s: String| -> Result<_, String> { libparsec::BackendOrganizationAddr::from_any(&s).map_err(|e| e.to_string()) }"
+    custom_to_rs_string = "|addr: libparsec::BackendOrganizationAddr| -> Result<String, &'static str> { Ok(addr.to_url().into()) }"
+
+
+class BackendOrganizationBootstrapAddr(StrBasedType):
+    custom_from_rs_string = "|s: String| -> Result<_, String> { libparsec::BackendOrganizationBootstrapAddr::from_any(&s).map_err(|e| e.to_string()) }"
+    custom_to_rs_string = "|addr: libparsec::BackendOrganizationBootstrapAddr| -> Result<String, &'static str> { Ok(addr.to_url().into()) }"
+
+
+class BackendInvitationAddr(StrBasedType):
+    custom_from_rs_string = "|s: String| -> Result<_, String> { libparsec::BackendInvitationAddr::from_any(&s).map_err(|e| e.to_string()) }"
+    custom_to_rs_string = "|addr: libparsec::BackendInvitationAddr| -> Result<String, &'static str> { Ok(addr.to_url().into()) }"
+
+
+class ParseBackendAddrError(Variant):
+    InvalidUrl = VariantItemUnit
```

Comment:
Use an enum ?

Reply:
Done

Reply:
good catch ! this should be an `ErrorVariant`, not a `Variant`

Reply:
Did you mean https://github.com/Scille/parsec-cloud/pull/5182#discussion_r1324359587 ?